### PR TITLE
feat: New design on producers platform

### PR DESCRIPTION
**Pull request by stephanegigandet**
This PR brings the new design to the producers platform.

Notable changes:
- New colorful design :)
- Product attributes shown on list of products + product summary page, according to user preferences
- Personal scoring of products based on users preferences (if enabled by the user)
- Knowledge panels on product pages

Also removed some dead code that was use in the old design (e.g. functions and templates used to generate blocks shown on the now defunct left sidebar).

### Screenshots

image

image

image

image

### CODE REVIEW

**Review by teolemon — APPROVED** [review]
- I added some screenshots
- We should have the key actions inline as buttons (Import file…), otherwise the CTR is going to decrease

**Comment by stephanegigandet** [comment]
> * We should have the key actions inline as buttons (Import file…), otherwise the CTR is going to decrease

We already have the "Import products" and "Export products" at the top of the front page:

image

We can add "Import photos" and turn the links to buttons.

**Comment by stephanegigandet** [comment]
Now with buttons:

image

File: templates/web/common/includes/producers_platform_top_panel.tt.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<!-- start templates/[% component.name %] -->
+
```
Comment (alexgarel) [inline templates/web/common/includes/producers_platform_top_panel.tt.html:2]: I think we should add more comments in templates using the `[%# comment %]`

```suggestion
[%# this template is included on home page, it helps redirect producers to pro platform %] 
```

File: templates/web/common/includes/producers_platform_top_panel.tt.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            </div>
+        </div>
+    </div>
+    [% END %]
+
+    [% IF !org_id.defined %]
+    <div class="row">
+        <div class="small-12 column v-space-short">
+            <div class="panel">
+                <p>[% lang('account_without_org') %]</p>
```
Comment (alexgarel) [inline templates/web/common/includes/producers_platform_top_panel.tt.html:34]: Don't we want to also have a link to public platform here, for users that may get lost ?

File: templates/web/common/includes/producers_platform_top_panel.tt.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<!-- start templates/[% component.name %] -->
+
+[% IF server_options_private_products.defined && server_options_private_products %]
+
+    [% IF pro_moderator %]
+    <div class="row">
```
Comment (alexgarel) [inline templates/web/common/includes/producers_platform_top_panel.tt.html:6]:
```suggestion
    [%# Form for moderators to switch org %]
    <div class="row">
```

File: templates/web/common/includes/producers_platform_menu_links.tt.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!-- start templates/[% template.name %] -->
+
+<li><label>[% lang("import_and_export_products") %]</label></li>
+<li><a href="/cgi/import_file_upload.pl">[% lang('import_product_data') %]</a></li>
+<li><a href="/cgi/import_photos_upload.pl">[% lang('import_product_photos') %]</a></li>
+<li><a href="/cgi/export_products.pl">[% lang('export_product_data_photos') %]</a></li>
+<li><a href="/cgi/remove_products.pl">[% lang('remove_products_from_producers_platform') %]</a></li>
+<li class="divider"></li>
```
Comment (alexgarel) [inline templates/web/common/includes/producers_platform_menu_links.tt.html:8]: A much wanted feature by @manon-corneille is a link to the user guide. Could we add it here ? (but it should be translatable string as the link differs between french and others (english for now)).

**Review by alexgarel — APPROVED** [review]
Really cool work.

I have some minor comments.

File: templates/web/common/includes/display_product_search_or_add_producer.tt.html [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
 
-<h3>[% lang("import_products") %]</h3>
-
 <p>
     &rarr; <a href="/cgi/import_file_upload.pl">[% lang('import_product_data') %]</a><br>
     &rarr; <a href="/cgi/import_photos_upload.pl">[% lang('import_product_photos') %]</a><br>
     &rarr; <a href="/cgi/export_products.pl">[% lang('export_product_data_photos') %]</a><br>
-    &rarr; <a href="/cgi/remove_products.pl">[% lang('remove_products_from_producers_platform') %]</a><br>
+    &rarr; <a href="/cgi/remove_products.pl">[% lang('remove_products_from_producers_platform') %]</a><b
+r>
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply templates/web/common/includes/display_product_search_or_add_producer.tt.html:8]: oops :) fixing

File: templates/web/common/includes/list_of_products.tt.html [hunk 6]
```diff
@@ -1,41 +1,44 @@
...
-      </li>
-      <li>
-        <a href="[% current_link_query_download %]&format=csv">[% lang('search_download_csv') %]</a>
-        ([% lang('search_download_csv_description') %])
-      </li>
-    </ul>
-    [% END %]
+    <div class="small-12 columns"> 
+      <p>&rarr;[% lang('search_download_results') %]</p>
+      [% IF count <= export_limit %]
```
[the remark this answers is not in the document]
Comment (stephanegigandet) [reply templates/web/common/includes/list_of_products.tt.html:27]: We display the limitation in the text:

```
msgctxt "search_download_results"
msgid "Download results in XLSX or CSV format. Please note that for performance reasons, you can download up to 10.000 results only."
msgstr "Download results in XLSX or CSV format. Please note that for performance reasons, you can download up to 10.000 results only."
```

We could restructure the downloads to make it possible to download the first 10k results, or to get pages or something, but currently it's not available.

File: templates/web/common/includes/producers_platform_top_panel.tt.html (hunk 1 above)
Comment (stephanegigandet) [reply templates/web/common/includes/producers_platform_top_panel.tt.html:2]: Good idea!

Comment (stephanegigandet) [reply templates/web/common/includes/producers_platform_top_panel.tt.html:2]: `[%# this template is included on home page, it helps redirect producers to pro platform %] 
`

In fact it does not work, it's currently not shown on the public platform.

File: templates/web/common/includes/producers_platform_menu_links.tt.html (hunk 4 above)
Comment (stephanegigandet) [reply templates/web/common/includes/producers_platform_menu_links.tt.html]: Good idea. I added a link to the user guide.

**Pull request merged**